hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/AbstractEvictableRecordStore.java

```diff
@@ -173,14 +167,19 @@ private int sampleForExpiry() {
     private int evictExpiredSamples(long now, boolean backup) {
         int evictedCount = 0;
 
-        Queue sampledPairs = SAMPLING_QUEUE.get();
-        Data key;
-        while ((key = (Data) sampledPairs.poll()) != null) {
-            Record record = (Record) sampledPairs.poll();
-            if (getOrNullIfExpired(key, record, now, backup) == null) {
-                evictedCount++;
+        List sampledPairs = SAMPLING_LIST.get();
+        try {
```

Comment:
I think this `try-finally` can be moved to `evictExpiredEntriesInternal()`. If `sampleForExpiry()` throws for any reason, we may still end up with some entries in the list.

Reply:
I moved try-finally in `evictExpiredSamples` because i need to free array-list to use it in the next iteration(since `evictExpiredEntriesInternal` can have more than 1 iteration). 

Added a specific try-catch block, after your comment, in `evictExpiredEntriesInternal`, in case of any exception now i clear thread local list.

Reply:
Thanks, LGTM :+1: